prog/mutation.go

```diff
@@ -320,7 +320,28 @@ func (t *IntType) mutate(r *randGen, s *state, arg Arg, ctx ArgCtx) (calls []*Ca
 }
 
 func (t *FlagsType) mutate(r *randGen, s *state, arg Arg, ctx ArgCtx) (calls []*Call, retry, preserve bool) {
-	return mutateInt(r, s, arg)
+	if !t.BitMask {
+		return mutateInt(r, s, arg)
+	}
+
+	a := arg.(*ConstArg)
```

Comment:
All of this logic should be in randGen.flags function: we want smarter handling of flags during both mutation and generation and we don't want it to be duplicated.
Extend the flags function to accept the current value of flags (generation will pass 0) and the BitMask flag.